h2o-py/demos/xgboost_feature_interactions_demo.ipynb

```diff
@@ -0,0 +1,1177 @@
+    "            stack.append((children_right[node_id], parent_depth + 1))\n",
+    "        else:\n",
+    "            is_leaves[node_id] = True\n",
+    "\n",
+    "    print(\"The binary tree structure has %s nodes and has \"\n",
+    "          \"the following tree structure:\"\n",
+    "          % n_nodes)\n",
+    "    for i in range(n_nodes):\n",
+    "        if is_leaves[i]:\n",
+    "            print(\"%snode=%s leaf node.\" % (node_depth[i] * \"\\t\", i))\n",
+    "        else:\n",
+    "            print(\"%snode=%s test node: go to node %s if X[:, %s] <= %s else to \"\n",
+    "                  \"node %s.\"\n",
+    "                  % (node_depth[i] * \"\\t\",\n",
+    "                     i,\n",
+    "                     children_left[i],\n",
+    "                     feature[i],\n",
+    "                     threshold[i],\n",
+    "                     children_right[i],\n",
+    "                     ))\n",
+    "    print()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Parse progress: |█████████████████████████████████████████████████████████| 100%\n",
+      "xgboost Model Build progress: |███████████████████████████████████████████| 100%\n"
+     ]
+    }
+   ],
+   "source": [
+    "from h2o.estimators.xgboost import *\n",
+    "# check if the H2O XGBoostEstimator is available\n",
+    "assert H2OXGBoostEstimator.available() is True\n",
+    "\n",
+    "# import data\n",
+    "data = h2o.import_file(path = \"../../smalldata/logreg/prostate.csv\")\n",
+    "\n",
+    "x = list(range(1, data.ncol-2))\n",
+    "y = data.names[len(data.names) - 1]\n",
+    "\n",
+    "ntree = 5\n",
+    "\n",
+    "h2o_params = {\n",
+    "    'eta': 0.3, \n",
+    "    'max_depth': 3,  \n",
+    "    'ntrees': ntree,\n",
+    "    'tree_method': 'hist'\n",
+    "} \n",
+    "\n",
+    "# define interactions as a list of list of names of colums\n",
+    "# the lists defines allowed interaction\n",
+    "# the interactions of each column with itself are always allowed\n",
+    "# so you cannot specified list with one column e.g. [\"PSA\"]\n",
+    "h2o_params[\"interaction_constraints\"] = [[\"CAPSULE\", \"AGE\"], [\"PSA\", \"DPROS\"]]\n",
+    "\n",
+    "# train h2o XGBoost model\n",
+    "h2o_model = H2OXGBoostEstimator(**h2o_params)\n",
+    "h2o_model.train(x=x, y=y, training_frame=data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Tree index:0\n",
+      "Node ID 0 has left child node with index 1 and right child node with index 2. The split feature is CAPSULE.\n",
+      "Node ID 1 has left child node with index -1 and right child node with index -1. The split feature is None.\n",
```

Comment:
Wouldn't it be better to call them "Leafs"?

Reply:
I think it is not important for this example, but it is not hard to change it.

Reply:
I see

Reply:
I improved the print to show leaf nodes better.